src/smart-components/role/RolesTable.tsx

```diff
@@ -86,12 +101,54 @@ const RolesTable: React.FunctionComponent<RolesTableProps> = ({ selectedRole })
     });
   }, [fetchData, page, perPage]);
 
-  const rows = useMemo(() => {
+  const getSortParams = (columnIndex: number): ThProps['sort'] => ({
+    sortBy: {
+      index: sortByIndex,
+      direction,
+      defaultDirection: 'asc',
+    },
+    onSort: (_event, index, direction) => onSort(_event, COLUMNHEADERS[index].key, direction),
+    columnIndex,
+  });
+
+  const sortData = (data: Role[], sortBy: string | undefined, direction: 'asc' | 'desc' | undefined) =>
+    sortBy && direction
+      ? [...data].sort((a, b) =>
+          direction === 'asc'
+            ? a[sortBy as keyof Role] < b[sortBy as keyof Role]
+              ? -1
+              : a[sortBy as keyof Role] > b[sortBy as keyof Role]
+              ? 1
+              : 0
+            : a[sortBy as keyof Role] > b[sortBy as keyof Role]
+            ? -1
+            : a[sortBy as keyof Role] < b[sortBy as keyof Role]
+            ? 1
+            : 0
+        )
+      : data;
+
+  const columns: DataViewTh[] = COLUMNHEADERS.map((column, index) => ({
+    cell: column.label,
+    props: { sort: getSortParams(index) },
+  }));
+
+  const filteredData = useMemo(
+    () =>
+      roles.filter(
+        (role) =>
+          (!filters.display_name || role.display_name?.toLocaleLowerCase().includes(filters.display_name?.toLocaleLowerCase())) &&
+          (!filters.description || role.description?.toLocaleLowerCase().includes(filters.description?.toLocaleLowerCase()))
+      ),
+    [filters, roles]
+  );
```

Comment:
The same here, we don't need to filter data on our side and rather fetch them from the API